docs/testnet-guardian-runbook.md

```diff
@@ -0,0 +1,130 @@
+# Testnet-Guardian Runbook
+
+## Overview
+
+A testnet guardian is the point of contact for anything engineering-related for an upcoming or a running testnet. Broadly speaking, a testnet guardian would perform/coordinate tasks involving building, testing and deploying testnet(s) for a release. They would also serve as a point-of-contact for protocol/product related queries from the community. This document describes all the tasks that a testnet guardian would perform during a release cycle and describes possible actions  in the case of events that affect the testnet. The role of a testnet guardian spans over a release cycle starting from code-freeze from the engineering team to the end of the release. The initial phase between code freeze and the public release is the QA phase where the testnet guardian will be responsible for building, deploying and testing internal QA-nets. On successful completion of the QA phase, the release phase starts during which the testnet guardian would support the release process and be the point of contact for the community.
+
+## QA-net
+
+The role of a testnet guardian for a release begins after the code-freeze. The following is the list of tasks that a testnet guardian would be responsible for:
+
+### Release branch
+
+Cut a release branch after the code-freeze(The naming convention for beta-testnets is `release/0.0.<release_number>-beta`. For example, `release/0.0.13-beta`)
+
+### Build
+
+The latest commit on the release branch should have successful builds for both macOS and Linux.
+Go to the CI workflow for the specific commit and look for `build- macos` and `build-artifacts--testnet_postake_medium_curves` jobs for mac and linux builds respectively.
+
+##### Update brew package
+To update brew package,
+
+1. Download the homebrew binary package from gcloud at the location specified in the `Copy artifacts to cloud` task in the `build-macOS` CI job. For example `gs://network-debug/382998/build/homebrew-coda.tar.gz`
+2. Upload it to our public s3 bucket at `https://s3-us-west-2.amazonaws.com/packages.o1test.net`
+3. In `coda.rb` in `CodaProtocol/homebrew-coda`,
+        1. update the url with the location of the new tarball
+        2. update the hash. The file containing the hash is uploaded to gcloud as well, for example `gs://network-debug/382998/build/homebrew-coda.tar.gz.sha256`
+        3. increment the revision number
```

Comment:
We were also baking the seed peers into the brew package though I guess we'll have to stop doing this if we want to run private testnets
https://github.com/CodaProtocol/homebrew-coda/blob/014b74bc84973910bb0a4857447f671b03621b24/coda.rb#L7-L8

Reply:
@Schmavery Ideally the user could override the baked-in peers

Reply:
Is this used? Seems like it has some old peer ids. We specify the initial peers in the daemon command on our docs page. Would that not be sufficient?

Reply:
@deepthiskumar this will be used in the 3.2b release (it is not used until we have a public network)

Reply:
We need it to support the `brew service start` flow